app/models/corrective_action.rb

```diff
@@ -28,12 +40,17 @@ class CorrectiveAction < ApplicationRecord
   validates :duration, inclusion: { in: DURATION_TYPES }, if: -> { duration.present? }
   validates :geographic_scope, presence: true
   validates :geographic_scope, inclusion: { in: Rails.application.config.corrective_action_constants["geographic_scope"] }, if: -> { geographic_scope.present? }
+  validates :action, inclusion: { in: actions.keys }
+  validates :other_action, presence: true, length: { maximum: 10_000 }, if: -> { action == "other" }
```

Comment:
Use the enum method here?

```suggestion
  validates :other_action, presence: true, length: { maximum: 10_000 }, if: -> { other? }
```